### ISSUE TITLE
[Bug]: Poor performance when inserting binary vectors

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#18985]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.1
- Deployment mode(standalone or cluster): cluster deployed by docker-compose, one data node, one query node
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Use the following script to insert binary vectors, 50000 entities per insert action:
```
def gen_binary_vectors(num, dim):
    raw_vectors = []
    binary_vectors = []
    for i in range(num):
        raw_vector = [random.randint(0, 1) for i in range(dim)]
        raw_vectors.append(raw_vector)
        binary_vectors.append(bytes(np.packbits(raw_vector, axis=-1).tolist()))
    return raw_vectors, binary_vectors

if __name__ == "__main__":
    connections.connect()
    print("========== BINARY VECTOR ============")
    dim = 512
    fields = [
        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
        FieldSchema(
            name="vec",
            dtype=DataType.BINARY_VECTOR,
            dim=dim,
        ),
    ]
    schema = CollectionSchema(fields)
    collection = Collection("AAA", schema, consistency_level="Strong")
    print("collection created")

    nb = 50000
    for i in range(40):
        print("Prepare data...")
        t1 = time.time()
        raw_vectors, binary_vectors = gen_binary_vectors(nb, dim=dim)
        entities = [
            binary_vectors
        ]
        t2 = time.time()
        print("Prepare", nb, "entities time cost", (t2 - t1), "seconds")
        print("Start insert")
        collection.insert(entities)
        t3 = time.time()
        print("Insert", nb, "entities time cost", (t3-t2), "seconds")

```
The time cost statistics:
```
========== BINARY VECTOR ============
collection created
Prepare data...
Prepare 50000 entities time cost 18.235697507858276 seconds
Start insert
Insert 50000 entities time cost 42.463125228881836 seconds
Prepare data...
Prepare 50000 entities time cost 18.571954488754272 seconds
Start insert
Insert 50000 entities time cost 9.314802169799805 seconds
Prepare data...
Prepare 50000 entities time cost 18.535671949386597 seconds
Start insert
Insert 50000 entities time cost 9.718725442886353 seconds
```

Then use the following script to insert float vectors, 5000 entities per insert action:
```
if __name__ == '__main__':
    connections.connect()
    print("========== FLOAT VECTOR ============")
    dim = 512
    fields = [
        FieldSchema(name="id", dtype=DataType.INT64, description="int64", is_primary=True, auto_id=True),
        FieldSchema(name="vec", dtype=DataType.FLOAT_VECTOR, dim=dim),
    ]
    schema = CollectionSchema(fields)
    collection = Collection("BBB", schema, consistency_level="Strong")
    print("collection created")

    nb = 50000
    for i in range(40):
        print("Prepare data...")
        t1 = time.time()
        entities = [
            [[random.random() for _ in range(dim)] for _ in range(nb)],
        ]
        t2 = time.time()
        print("Prepare", nb, "entities time cost", (t2 - t1), "seconds")
        print("Start insert")
        collection.insert(entities)
        t3 = time.time()
        print("Insert", nb, "entities time cost", (t3 - t2), "seconds")
```

The time cost statistics:
```
========== FLOAT VECTOR ============
collection created
Prepare data...
Prepare 50000 entities time cost 2.759742498397827 seconds
Start insert
Insert 50000 entities time cost 4.04320764541626 seconds
Prepare data...
Prepare 50000 entities time cost 2.902754306793213 seconds
Start insert
Insert 50000 entities time cost 3.977565050125122 seconds
Prepare data...
Prepare 50000 entities time cost 3.157877206802368 seconds
Start insert
Insert 50000 entities time cost 4.011607646942139 seconds
```

The performance of inserting binary vectors is poor, compare to float vectors.

### Expected Behavior

With the same dimension, binary vector size is less than float vector. Then inserting binary vectors should not slow than inserting float vectors.

**Comment by yhmo** [comment]
The bottleneck is in Pymilvus SDK side.
In the pymilvus/client/prepare.py, 
`def bulk_insert_param(cls, collection_name, entities, partition_name, fields_info=None, **kwargs)`

**Comment by yhmo** [comment]
The root cause:
concatenate bytes in Python is very slow

https://www.guyrutenberg.com/2020/04/04/fast-bytes-concatenation-in-python/

**Comment by yhmo** [comment]
This bug has been fixed in pymilvus: https://github.com/milvus-io/pymilvus/pull/1134
The fix will be available in the next minio version of pymilvus.

**Comment by yanliang567** [comment]
/assign @binbinlv 
could you please help to verify the fix?

/unassign

**Comment by binbinlv** [comment]
OK, verifying

**Comment by binbinlv** [comment]
Verified.

In pymilvus: 2.2.0.dev26:

```
========== BINARY VECTOR ============
collection created
Prepare data...
Prepare 50000 entities time cost 17.92335557937622 seconds
Start insert
Insert 50000 entities time cost 44.401010036468506 seconds
Prepare data...
Prepare 50000 entities time cost 17.844159603118896 seconds
Start insert
Insert 50000 entities time cost 17.939941883087158 seconds
Prepare data...
Prepare 50000 entities time cost 17.87653636932373 seconds
Start insert
Insert 50000 entities time cost 19.60693335533142 seconds
Prepare data...
Prepare 50000 entities time cost 17.510582208633423 seconds
Start insert
Insert 50000 entities time cost 17.971370935440063 seconds
Prepare data...
Prepare 50000 entities time cost 17.92458415031433 seconds
Start insert
Insert 50000 entities time cost 18.239166498184204 seconds
Prepare data...
Prepare 50000 entities time cost 17.754257678985596 seconds
```

Using the fixed pymilvus version: 2.2.0.dev27
```
========== BINARY VECTOR ============
collection created
Prepare data...
Prepare 50000 entities time cost 18.369293212890625 seconds
Start insert
Insert 50000 entities time cost 0.09581899642944336 seconds
Prepare data...
Prepare 50000 entities time cost 18.286952257156372 seconds
Start insert
Insert 50000 entities time cost 0.0555264949798584 seconds
Prepare data...
Prepare 50000 entities time cost 18.41678524017334 seconds
Start insert
Insert 50000 entities time cost 0.0613861083984375 seconds
Prepare data...
Prepare 50000 entities time cost 18.23392677307129 seconds
Start insert
Insert 50000 entities time cost 0.05830073356628418 seconds
Prepare data...
Prepare 50000 entities time cost 18.318632125854492 seconds
Start insert
Insert 50000 entities time cost 0.05269503593444824 seconds
Prepare data...
Prepare 50000 entities time cost 18.378522634506226 seconds
Start insert
Insert 50000 entities time cost 0.06045722961425781 seconds
```
